src/collection-view.js

```diff
@@ -281,17 +275,25 @@ const CollectionView = Backbone.View.extend({
     if (anyModelsAdded) {
       this.render();
     } else {
-      // Get the DOM nodes in the same order as the models.
-      const elsToReorder = _.map(models, function(model, index) {
-        const view = children.findByModel(model);
+
+      const filteredOutModels = [];
+
+      // Get the DOM nodes in the same order as the models and
+      // find the model that were children before but aren't in this new ordering.
+      const elsToReorder = children.reduce(function(viewEls, view) {
+        const index = _.indexOf(models, view.model);
+
+        if (index === -1) {
+          filteredOutModels.push(view.model);
+          return viewEls;
+        }
+
         view._index = index;
-        return view.el;
-      });
 
-      // Find the views that were children before but aren't in this new ordering.
-      const filteredOutViews = children.filter(function(view) {
-        return !_.contains(elsToReorder, view.el);
-      });
+        viewEls[index] = view.el;
+
+        return viewEls;
+      }, new Array(models.length));
```

Comment:
I'm pretty sure it will match.. if there are models not in the children it wouldn't get into this `else`.. TBH this whole function gives me the heeby-jeebies.